lib/scholar/neighbors/random_projection_forest.ex

```diff
@@ -336,13 +326,16 @@ defmodule Scholar.Neighbors.RandomProjectionForest do
       |> Nx.vectorize(:trees)
       |> Nx.add(start_indices)
 
-    candidate_indices =
-      Nx.take(indices, pos)
-      |> Nx.devectorize()
-      |> Nx.rename(nil)
-      |> Nx.transpose(axes: [1, 0, 2])
-      |> Nx.reshape({query_size, num_trees * leaf_size})
+    Nx.take(indices, pos)
+    |> Nx.devectorize()
+    |> Nx.rename(nil)
+    |> Nx.transpose(axes: [1, 0, 2])
+    |> Nx.reshape({query_size, num_trees * leaf_size})
+  end
 
+  defnp predict_n(forest, query) do
+    k = forest.num_neighbors
+    candidate_indices = get_leaves(forest, query)
```

Comment:
I would move `get_leaves` below `predict_n` as the code seems more understandable to me. This is just a personal preference; I am not aware of any conventions imposing this.

Reply:
Sure